doc/source/drivers/raster/gdalg.rst

```diff
@@ -0,0 +1,64 @@
+.. _raster.gdalg:
+
+================================================================================
+GDALG: GDAL Streamed Algorithm
+================================================================================
+
+.. versionadded:: 3.11
+
+.. shortname:: GDALG
+
+.. built_in_by_default::
+
+This is a read-only driver that reads a JSON file containing a command line
+of the :ref:`gdal command line interface <gdal_program>`, that results in a
+on-the-fly / streamed raster dataset when this file is used as a input dataset
+for :cpp:func:`GDALOpenEx`, or anywhere in GDAL API or command line utilities
+where a raster input dataset is expected. GDALG files are conceptually close
+to :ref:`VRT (Virtual) files <raster.vrt>`, although the implementation is
+substantially different.
+
+The subset of commands of :program:`gdal` that can generate such streamed datasets
+indicates ``GDALG`` is a valid output format. This includes :ref:`gdal_raster_pipeline_subcommand`,
+:ref:`gdal_raster_mosaic_subcommand` or :ref:`gdal_raster_stack_subcommand`.
+
+It is recommended that GDALG files use the ``.gdalg`` extension.
```

Comment:
I'm guessing this is to facilitate detection? (I think `.gdalg.json` would be friendlier for users trying to edit these files)

Reply:
Yes, it's my experience hand-editing `.vrt` files that makes me wonder about `gdalalg.json` :)

Reply:
extension changed to ``.gdalg.json``